framework/Samplers/MCMC/MCMC.py

```diff
@@ -0,0 +1,293 @@
+    samplerInitInput.addSub(limitInput)
+    initialSeedInput = InputData.parameterInputFactory("initialSeed", contentType=InputTypes.IntegerType,
+        descr='')
+    samplerInitInput.addSub(initialSeedInput)
+    tuneInput = InputData.parameterInputFactory("tune", contentType=InputTypes.IntegerType,
+        descr='')
+    samplerInitInput.addSub(tuneInput)
+    inputSpecification.addSub(samplerInitInput)
+
+    inputSpecification.addSub(InputData.parameterInputFactory("likelihood",contentType=InputTypes.StringType,
+        printPriority=5,
+        descr=r"""Output of likelihood function"""))
+    # modify Sampler variable nodes
+    variable = inputSpecification.getSub('variable')
+    variable.addSub(InputData.parameterInputFactory('initial', contentType=InputTypes.FloatType,
+        descr=r"""inital value for given variable"""))
+    variable.addSub(InputData.assemblyInputFactory('proposal', contentType=InputTypes.StringType, strictMode=True,
+        printPriority=30,
+        descr=r"""name of the Distribution that is used as proposal distribution"""))
+
+    inputSpecification.addSub(variable)
+    return inputSpecification
+
+  @classmethod
+  def getSolutionExportVariableNames(cls):
+    """
+      Compiles a list of acceptable SolutionExport variable options.
+      @ In, None
+      @ Out, vars, dict, {varName: manual description} for each solution export option
+    """
+    vars = super(AdaptiveSampler, cls).getSolutionExportVariableNames()
+    # TODO: multiple chains for MCMC
+    new = {'traceID': 'integer identifying which iteration a Markov chain is on',
+           '{VAR}': r'any variable from the \xmlNode{TargetEvaluation} input or output at current iteration'
+           }
+    vars.update(new)
+    return vars
+
+  def __init__(self):
+    """
+      Default Constructor that will initialize member variables with reasonable
+      defaults or empty lists/dictionaries where applicable.
+      @ In, None
+      @ Out, None
+    """
+    AdaptiveSampler.__init__(self)
+    self._initialValues = {} # dict stores the user provided initial values, i.e. {var: val}
+    self._updateValues = {} # dict stores input variables values for the current MCMC iteration, i.e. {var:val}
+    self._proposal = {} # dict stores the proposal distributions for input variables, i.e. {var:dist}
+    self._tune = 0      # integers indicate how many samples will be discarded
+    self._likelihood = None # stores the output from the likelihood
+    self._availProposal = {'normal': Distributions.Normal(0.0, 1.0),
+                           'uniform': Distributions.Uniform(-1.0, 1.0)} # available proposal distributions
+    self._acceptDist = Distributions.Uniform(0.0, 1.0) # uniform distribution for accept/rejection purpose
+    # assembler objects
+    self.addAssemblerObject('proposal', '-n', True)
+
+  def localInputAndChecks(self, xmlNode, paramInput):
+    """
+      unfortunately-named method that serves as a pass-through for input reading.
```

Comment:
unfortunately-named method? O_o

Reply:
10:1 says I wrote the original version this was copied from ... heh

Reply:
fixed